core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1686,12 +1689,26 @@ private void processBind(long lo, long msgLimit, @Transient SqlCompiler compiler
         validateParameterCounts(parameterFormatCount, parameterValueCount, parsePhaseBindVariableCount);
 
         lo += Short.BYTES;
-        if (parameterValueCount > 0) {
-            if (this.parsePhaseBindVariableCount == parameterValueCount) {
-                lo = bindValuesUsingSetters(lo, msgLimit, parameterValueCount);
-            } else {
-                lo = bindValuesAsStrings(lo, msgLimit, parameterValueCount);
+
+        try {
+            if (parameterValueCount > 0) {
+                if (this.parsePhaseBindVariableCount == parameterValueCount) {
+                    lo = bindValuesUsingSetters(lo, msgLimit, parameterValueCount);
+                } else {
+                    lo = bindValuesAsStrings(lo, msgLimit, parameterValueCount);
+                }
             }
+        } catch (SqlException e) {
+            //if statement is named or portal then we've to add it to typesAndSElect 
+            if (typesAndSelect != null &&
```

Comment:
I think, the branch condition could be simplified to just `typesAndSelect != null`. That's because `typesAndSelect` is only initialized via a `typesAndSelectCache.poll()` call.

Reply:
If we don't check cache we might end up removing element already in cache and that might explode on next prepared statement call . 
To be honest we should rather keep the query even if it fails in bind phase because it's already been compiled and could be used in later in the 'session' so this piece only serves as a temporary bandaid .

Reply:
The thing is that the element is not in the cache and that's the `typesAndSelect != null` invariant. If the condition is `true`, it's not in the cache. Other parts of this class already rely on that invariant, so these additional check just do nothing.